connectivity/perf/benchmarks/netperf/perfpod.go

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package netperf
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cilium/cilium-cli/connectivity/check"
+	"github.com/cilium/cilium-cli/connectivity/perf/common"
+	"github.com/cilium/cilium-cli/utils/features"
+)
+
+const (
+	messageSize = 1024
+	tool        = "netperf"
+)
+
+// Network Performance
+func Netperf(n string) check.Scenario {
+	return &netPerf{
+		name: n,
+	}
+}
+
+type netPerf struct {
+	name string
+}
+
+func (s *netPerf) Name() string {
+	if s.name == "" {
+		return tool
+	}
+	return fmt.Sprintf("%s:%s", tool, s.name)
+}
+
+func (s *netPerf) Run(ctx context.Context, t *check.Test) {
+	samples := t.Context().Params().PerfSamples
+	duration := t.Context().Params().PerfDuration
+
+	for sample := 1; sample <= samples; sample++ {
+		for _, c := range t.Context().PerfClientPods() {
+			c := c
+			for _, server := range t.Context().PerfServerPod() {
+				scenarioName := ""
+				if strings.Contains(c.Pod.Name, check.PerfHostName) {
+					scenarioName += "host"
+				} else {
+					scenarioName += "pod"
+				}
+				scenarioName += "-to-"
+				if strings.Contains(server.Pod.Name, check.PerfHostName) {
+					scenarioName += "host"
+				} else {
+					scenarioName += "pod"
+				}
+				sameNode := true
+				if strings.Contains(c.Pod.Name, check.PerfOtherNode) {
+					sameNode = false
+				}
+
+				tests := []string{
```

Comment:
Do we need to redefine this test slice every single time?

Reply:
Fixed.